tutorials/scripting/gdextension/gdextension_cpp_example.rst

```diff
@@ -0,0 +1,652 @@
+.. _doc_gdextension_cpp_example:
+
+GDExtension C++ example
+=======================
+
+Introduction
+------------
+
+The C++ bindings for GDExtension are built on top of the C GDExtension API
+and provide a nicer way to "extend" nodes in Godot using C++.
+This new system allows the extension of Godot to nearly the same
+level as statically linked C++ modules.
+
+You can download the included example in the test folder of the godot-cpp
+repository `on GitHub <https://github.com/godotengine/godot-cpp>`__.
+
+Setting up the project
+----------------------
+
+There are a few prerequisites you'll need:
+
+- a Godot 4 executable,
+- a C++ compiler,
+- SCons as a build tool,
+- a copy of the `godot-cpp
+  repository <https://github.com/godotengine/godot-cpp>`__.
+
+See also :ref:`Compiling <toc-devel-compiling>` as the build tools are identical
+to the ones you need to compile Godot from source.
+
+You can download this repository from GitHub or let Git do the work for you.
+Note that this repository has different branches for different versions
+of Godot. GDExtensions will not work in older versions of Godot (only Godot 4 and up) and vice versa, so make sure you download the correct branch.
+
+.. note::
+
+    `GDExtension <https://godotengine.org/article/introducing-gd-extensions>`__
+    has been merged in the ``master`` branch of godot-cpp,
```

Comment:
Hmm, this note is weird. It makes sense to have in the Godot 3 documentation to refer to needing to use a branch other than master in order to still do things with GDNative.

But I think using the master branch of godot-cpp is expected especially with a view of Godot 4 stable being released any day now.

I'd remove it or only keep `You need to use the ``master`` branch of godot-cpp to use GDExtension and follow this example.`